Update salesforce docs to include recycle bin info

katmarkham opened the pull request:
Updating the docs to be more clear on how deleted records and the salesforce recycle bin work within our streams.

katmarkham committed:
Update salesforce.md to include recycle bin info

Updating the docs to be more clear on how deleted records and the salesforce recycle bin work within our streams.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=30420) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=30420) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=30420) it.</sub>

katmarkham committed:
Update salesforce.inapp.md to include details on Salesforce Recycle Bin

katmarkham committed:
Update salesforce.inapp.md

Just realized that the "For detailed information" section right above the fold was directing to Google Analytics 4 instead of Salesforce

katmarkham committed:
Merge branch 'master' into katmarkham/salesforce

katmarkham committed:
Merge branch 'master' into katmarkham/salesforce
